contracts/test/SimplePDPService.t.sol

```diff
@@ -78,4 +78,100 @@ contract SimplePDPServiceTest is Test {
         vm.expectRevert("Event index out of bounds");
         pdpService.getEvent(proofSetId, 0);
     }
+
+    function testGetMaxProvingPeriod() public view {
+        uint64 maxPeriod = pdpService.getMaxProvingPeriod();
+        assertEq(maxPeriod, 2880, "Max proving period should be 2880");
+    }
+
+    function testGetChallengesPerProof() public view{
+        uint64 challenges = pdpService.getChallengesPerProof();
+        assertEq(challenges, 5, "Challenges per proof should be 5");
+    }
+}
+
+contract SimplePDPServiceFaultsTest is Test {
+    SimplePDPService public pdpService;
+    address public pdpVerifierAddress;
+
+    function setUp() public {
+        pdpVerifierAddress = address(this);
+        SimplePDPService pdpServiceImpl = new SimplePDPService();
+        bytes memory initializeData = abi.encodeWithSelector(SimplePDPService.initialize.selector, address(pdpVerifierAddress));
+        MyERC1967Proxy pdpServiceProxy = new MyERC1967Proxy(address(pdpServiceImpl), initializeData);
+        pdpService = SimplePDPService(address(pdpServiceProxy));
+    }
+
+    function testPosessionProvenOnTime() public {
+        uint256 proofSetId = 1;
+        uint256 challengedLeafCount = 100;
+        uint256 seed = 12345;
+        uint256 challengeCount = 5;
+
+        // Set up the proving deadline
+        pdpService.rootsAdded(proofSetId, 0, new PDPVerifier.RootData[](0));
+        // Warp to just before the deadline
+        vm.warp(block.number + pdpService.getMaxProvingPeriod() - 1);
+        pdpService.posessionProven(proofSetId, challengedLeafCount, seed, challengeCount);
```

Comment:
I believe the proof should also be accepted _on_ the deadline, so exercise that here.